packages/apollos-data-prayer/src/prayer-requests/data-source.js

```diff
@@ -94,26 +94,28 @@ export default class PrayerRequest extends RockApolloDataSource {
     }
   };
 
-  // QUERY PrayerRequests by Campus
-  getAllByCampus = async (campusId) => {
-    try {
-      const {
-        dataSources: { Auth },
-      } = this.context;
+  getAllByCampus = async (id = '') => {
+    const {
+      dataSources: { Auth, Campus },
+    } = this.context;
 
-      const { primaryAliasId } = await Auth.getCurrentPerson();
+    const { primaryAliasId } = await Auth.getCurrentPerson();
 
-      const prayers = await this.request('PrayerRequests/Public')
-        .filter(
-          `(CampusId eq ${
-            parseGlobalId(campusId).id
-          }) and (RequestedByPersonAliasId ne ${primaryAliasId})`
-        )
-        .get();
-      return this.sortPrayers(prayers);
-    } catch (err) {
-      throw new Error(err);
+    let campus;
+    let campusID;
+    if (id === '') {
+      campus = await Campus.getForPerson(primaryAliasId);
+      campusID = campus.id;
+    } else {
+      campusID = parseGlobalId(id).id;
     }
+
+    const prayers = await this.request('PrayerRequests/Public')
+      .filter(
+        `(CampusId eq ${campusID}) and (RequestedByPersonAliasId ne ${primaryAliasId})`
+      )
+      .get();
+    return this.sortPrayers(prayers);
```

Comment:
You removed the `try ... catch`. How do we handle if the query throws an error?

Reply:
what's the difference in our "error handler" throwing an error or the exception itself? it wasn't doing anything.

Reply:
I'm not sure what you mean by "our error handler". I also don't really understand how our error handlers work, but here's my concern. I think it makes sense to be as specific as possible with our error handling. If this function errors, then I want to know that this function was the problem. What I don't want (and what is not helpful) is just getting some generic `GraphQLError` like we currently get in Bugsnag and then having to search through those to try to find the 1 that is the problem. If our errors are specific, then removing this is not a problem (I guess). If something generic bubbles up, then we need to do a better job of being more specific and less nonchalant with our errors. Does that make sense?